tonic/src/transport/server/mod.rs

```diff
@@ -330,18 +297,17 @@ impl<L> Server<L> {
     ///
     /// This will clone the `Server` builder and create a router that will
     /// route around different services.
-    pub fn add_service<S>(&mut self, svc: S) -> Router<S, Unimplemented, L>
+    pub fn add_service<S>(&mut self, svc: S) -> Router<L>
     where
-        S: Service<Request<Body>, Response = Response<BoxBody>>
+        S: Service<Request<Body>, Response = Response<BoxBody>, Error = Infallible>
```

Comment:
Should we do this in this PR or another but we should maybe make`Request<impl Body>` instead of hard coding it to hyper?

Reply:
Good idea. I think we should explore it separately though.